src/networking.c

```diff
@@ -4100,7 +4100,7 @@ void clientTrackingInfoCommand(client *c) {
 
     /* Prefixes */
     addReplyBulkCString(c, "prefixes");
-    if (c->pubsub_data->client_tracking_prefixes) {
+    if (c->pubsub_data && c->pubsub_data->client_tracking_prefixes) {
```

Comment:
This fix looks correct.

There are more occurrences of `c->pubsub_data->client_tracking_redirection` where we don't check `c->pubsub_data` first, for example 7 lines above this line, but in those cases we check `c->flag.tracking` first. Can we rely on this being initialized when the tracking flag is set?

If yes, then should we just check `c->flag.tracking` here too?

```suggestion
    if (c->flag.tracking) {
        serverAssert(c->pubsub_data && c->pubsub_data->client_tracking_prefixes);
```

Reply:
> If yes, then should we just check c->flag.tracking here too?

Yes, I think we can use `c->flag.tracking`. when `enableTracking`, will set c->flag.tracking = 1 and initClientPubSubData.